core/jobs/src/main/java/com/cognifide/aet/job/common/collectors/source/SourceCollector.java

```diff
@@ -84,23 +73,11 @@ public final CollectorStepResult collect() throws ProcessingException {
   }
 
   private byte[] getContent() throws ProcessingException {
-    byte[] content;
-    ExecutorService executor = Executors.newCachedThreadPool();
-    Callable<Object> task = new Callable<Object>() {
-      @Override
-      public Object call() throws IOException {
-        return httpRequestBuilder.executeRequest().getContent();
-      }
-    };
-    Future<Object> future = executor.submit(task);
     try {
-      content = (byte[]) future.get(timeoutValue, TimeUnit.MILLISECONDS);
-    } catch (TimeoutException | InterruptedException | ExecutionException e) {
+      return httpRequestExecutor.executeGetRequest(properties.getUrl(), timeoutValue).getContent();
+    } catch (IOException e) {
       throw new ProcessingException(e.getMessage(), e);
```

Comment:
Maybe we could add error LOG with exception message and requested URL?

Reply:
This is already logged in `CollectorDispatcherImpl` in the catch block.